src/CoreWCF.Primitives/src/CoreWCF/Security/FederatedMessageSecurityOverHttp.cs

```diff
@@ -0,0 +1,232 @@
+            issuedParameters.AddAlgorithmParameters(AlgorithmSuite, versionSpecificStandardsManager, _issuedKeyType);
+
+            SecurityBindingElement issuedTokenSecurity;
+            if (isSecureTransportMode)
+            {
+                issuedTokenSecurity = SecurityBindingElement.CreateIssuedTokenOverTransportBindingElement(issuedParameters);
+            }
+            else
+            {
+                if (NegotiateServiceCredential)
+                {
+                    // We should have passed 'true' as RequireCancelation to be consistent with other standard bindings.
+                    // However, to limit the change for Orcas, we scope down to just newer version of WSSecurityPolicy.
+                    issuedTokenSecurity = SecurityBindingElement.CreateIssuedTokenForSslBindingElement(issuedParameters, version.SecurityPolicyVersion != SecurityPolicyVersion.WSSecurityPolicy11);
+                }
+                else
+                {
+                    issuedTokenSecurity = SecurityBindingElement.CreateIssuedTokenForCertificateBindingElement(issuedParameters);
+                }
+            }
+
+            issuedTokenSecurity.MessageSecurityVersion = version;
+            issuedTokenSecurity.DefaultAlgorithmSuite = AlgorithmSuite;
+
+            if (EstablishSecurityContext)
+            {
+                result = SecurityBindingElement.CreateSecureConversationBindingElement(issuedTokenSecurity, true);
+            }
+            else
+            {
+                result = issuedTokenSecurity;
+            }
+
+            result.MessageSecurityVersion = version;
+            result.DefaultAlgorithmSuite = AlgorithmSuite;
+            result.IncludeTimestamp = true;
+
+            if (!isReliableSession)
+            {
+                result.LocalServiceSettings.ReconnectTransportOnFailure = false;
+            }
+            else
+            {
+                result.LocalServiceSettings.ReconnectTransportOnFailure = true;
+            }
+
+            if (EstablishSecurityContext)
+            {
+                // issue the transition SCT for a short duration only
+                issuedTokenSecurity.LocalServiceSettings.IssuedCookieLifetime = SpnegoTokenAuthenticator.s_defaultServerIssuedTransitionTokenLifetime;
+            }
+
+            return result;
+        }
+
+        internal bool InternalShouldSerialize()
+        {
+            return (ShouldSerializeAlgorithmSuite()
+                || ShouldSerializeClaimTypeRequirements()
+                || ShouldSerializeNegotiateServiceCredential()
+                || ShouldSerializeEstablishSecurityContext()
+                || ShouldSerializeIssuedKeyType()
+                || ShouldSerializeTokenRequestParameters());
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool ShouldSerializeAlgorithmSuite()
+        {
```

Comment:
Remove all the ShouldSerializeXXXX methods, they are only used by the .NET SDK tool svcconfigeditor.